pyqtgraph/parametertree/Parameter.py

```diff
@@ -39,11 +50,17 @@ def registerParameterType(name, cls, override=False):
     PARAM_TYPES[name] = cls
     PARAM_NAMES[cls] = name
 
+
 def __reload__(old):
     PARAM_TYPES.update(old.get('PARAM_TYPES', {}))
     PARAM_NAMES.update(old.get('PARAM_NAMES', {}))
 
-class Parameter(QtCore.QObject):
+
+class _ParameterMeta(ABCMeta, type(QtCore.QObject)):
```

Comment:
Do we really want to use metaclasses here and not something like `typing.Protocol`?

Reply:
I had some trouble combining `QObject` inheritance and abstract classes. It might be possible without an explicit metaclass, though, and I just couldn't figure it out.

Reply:
I haven't used metaclasses that much, I know they're somewhat more ....controversial in python-land than in other languages.

@ksunden you have any thoughts on the use here?

Reply:
no need; the path to being abstract has been abandoned :)